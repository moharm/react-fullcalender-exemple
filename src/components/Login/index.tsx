import React, { useContext, useEffect } from "react";
import CssBaseline from "@material-ui/core/CssBaseline";
import Container from "@material-ui/core/Container";
import TextField from "@material-ui/core/TextField";
import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import { RouteComponentProps } from "@reach/router";
import { AuthContext } from "../../contexts/AuthContext";
import {
  WithLastLocationProps,
  withLastLocation,
  useLastLocation
} from "react-router-last-location";
import history from "../../assets/history/history";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
      overflow: "hidden",
      padding: theme.spacing(0, 3),
      textAlign: "center"
    },
    input: {
      margin: theme.spacing(2)
    },
    button: {
      margin: theme.spacing(2)
    }
  })
);

const Login = (props: RouteComponentProps & WithLastLocationProps) => {
  const classes = useStyles();
  const { isAuthentificated, userName, password, Login, onChange } = useContext(
    AuthContext
  );
  const lastLocation = useLastLocation();

  const LastLocation = lastLocation ? lastLocation.pathname : "";

  useEffect(() => {
    if (isAuthentificated) {
      LastLocation ? history.push(LastLocation) : history.push("/App/home");
    }
  });

  return (
    <React.Fragment>
      <CssBaseline />
      <Container maxWidth="xs">
        <h1>{LastLocation}</h1>

        <form
          className={classes.root}
          autoComplete="off"
          onSubmit={event => Login(event, LastLocation)}
        >
          <div>
            <TextField
              className={classes.input}
              id="username"
              label="Username"
              name="userName"
              type="text"
              variant="outlined"
              value={userName}
              onChange={onChange}
              required
            />
            <TextField
              className={classes.input}
              id="password-input"
              label="Password"
              name="password"
              type="password"
              autoComplete="current-password"
              variant="outlined"
              value={password}
              onChange={onChange}
              required
            />
            <Button
              variant="contained"
              color="primary"
              className={classes.button}
              type="submit"
            >
              Login
            </Button>
          </div>
        </form>
      </Container>
    </React.Fragment>
  );
};
export default withLastLocation(Login);
