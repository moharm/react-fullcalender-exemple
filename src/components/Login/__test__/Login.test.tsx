import React from "react";
import { mount, shallow, render, ShallowWrapper } from "enzyme";
import Login from "..";
import { Router } from "react-router";
import history from "../../../assets/history/history";
import "@testing-library/jest-dom/extend-expect";
import TextField from "@material-ui/core/TextField";
import AuthContextProvider from "../../../contexts/AuthContext";

it("should display a blank login form", () => {
  const wrapper = mount(
    <AuthContextProvider>
      <Router history={history}>
        <Login />
      </Router>
    </AuthContextProvider>
  );
  expect(wrapper).toMatchSnapshot();
  const FORM = wrapper.find("form").first();
});

// function getByCssSelector(shallowComp: ShallowWrapper, keyVal)

it("renders username input with label (default type)", () => {
  const FORM = mount(
    <AuthContextProvider>
      <Router history={history}>
        <Login />
      </Router>
    </AuthContextProvider>
  );
  const username = FORM.find("#username").first();
  expect(username).toHaveLength(1);
  expect(username.prop("id")).toEqual("username");
  expect(username.prop("label")).toEqual("Username");
  expect(username.prop("name")).toEqual("userName");
  expect(username.prop("type")).toEqual("text");
});

it("renders password input with label given the type", () => {
  const FORM = mount(
    <AuthContextProvider>
      <Router history={history}>
        <Login />
      </Router>
    </AuthContextProvider>
  );
  const password = FORM.find("#password-input").first();
  expect(password).toHaveLength(1);
  expect(password.prop("id")).toEqual("password-input");
  expect(password.prop("label")).toEqual("Password");
  expect(password.prop("name")).toEqual("password");
  expect(password.prop("type")).toEqual("password");
  expect(password.prop("autoComplete")).toEqual("current-password");
  expect(password.prop("variant")).toEqual("outlined");
  expect(password.prop("value")).toEqual("");
});

// it("should allow entering a username", async () => {
//   const FORM = mount(
//     <AuthContextProvider>
//       <Router history={history}>
//         <Login />
//       </Router>
//     </AuthContextProvider>
//   );
//   const username = FORM.find("#username").first();

//   username.simulate("change", { value: "mohcine" });
//   username.update();
//   console.log(username.find("input").props());
//   expect(username.find("input").props().value).toEqual("mohcine");
// });
