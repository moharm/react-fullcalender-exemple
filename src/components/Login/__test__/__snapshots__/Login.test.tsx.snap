// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a blank login form 1`] = `ReactWrapper {}`;
